tests/Unit/Adapter/News/NewsDataProviderTest.php

```diff
@@ -0,0 +1,196 @@
+            $this->createCircuitBreakerMock(),
+            $this->createCountryDataProviderMock(),
+            $this->createToolsMock(),
+            $this->createConfigurationMock(),
+            $this->createValidateMock(),
+            ContextCore::MODE_STD
+        );
+        $this->assertNotNull($generator);
+    }
+
+    public function testDataFallback()
+    {
+        $expectedLocale = 'fr';
+
+        $generator = new NewsDataProvider(
+            $this->createCircuitBreakerMock($expectedLocale, ''),
+            $this->createCountryDataProviderMock(),
+            $this->createToolsMock(),
+            $this->createConfigurationMock(),
+            $this->createValidateMock(),
+            ContextCore::MODE_STD
+        );
+        $expectedJson = $generator->getData($expectedLocale);
+        $this->assertInternalType('array', $expectedJson);
+        $this->assertArrayHasKey('has_errors', $expectedJson);
+        $this->assertTrue($expectedJson['has_errors']);
+        $this->assertArrayHasKey('rss', $expectedJson);
+        $this->assertInternalType('array', $expectedJson['rss']);
+        $this->assertEmpty($expectedJson['rss']);
+    }
+
+    public function testDataRssInvalid()
+    {
+        $expectedLocale = 'fr';
+
+        $generator = new NewsDataProvider(
+            $this->createCircuitBreakerMock($expectedLocale, 'INVALID DATA'),
+            $this->createCountryDataProviderMock(),
+            $this->createToolsMock(),
+            $this->createConfigurationMock(),
+            $this->createValidateMock(),
+            ContextCore::MODE_STD
+        );
+        $expectedJson = $generator->getData($expectedLocale);
+        $this->assertInternalType('array', $expectedJson);
+        $this->assertArrayHasKey('has_errors', $expectedJson);
+        $this->assertTrue($expectedJson['has_errors']);
+        $this->assertArrayHasKey('rss', $expectedJson);
+        $this->assertInternalType('array', $expectedJson['rss']);
+        $this->assertEmpty($expectedJson['rss']);
+    }
+
+    public function testData()
+    {
+        $expectedLocale = 'fr';
+
+        $generator = new NewsDataProvider(
+            $this->createCircuitBreakerMock($expectedLocale, file_get_contents(__DIR__ . '/../../Resources/rss/blog-fr.xml')),
+            $this->createCountryDataProviderMock(),
+            $this->createToolsMock(),
+            $this->createConfigurationMock(),
+            $this->createValidateMock(true),
+            ContextCore::MODE_STD
+        );
+        $expectedJson = $generator->getData($expectedLocale);
+
+        $this->assertInternalType('array', $expectedJson);
+        $this->assertArrayHasKey('has_errors', $expectedJson);
+        $this->assertFalse($expectedJson['has_errors']);
+        $this->assertArrayHasKey('rss', $expectedJson);
+        $this->assertInternalType('array', $expectedJson['rss']);
+        $this->assertCount(NewsDataProvider::NUM_ARTICLES, $expectedJson['rss']);
```

Comment:
you should test the format of the items you return, and maybe check the url parameters, since the most complex thing in the provider is building the url with campaign parameters (just checking they are here should be enough)

Reply:
:ok: